packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -623,195 +759,180 @@ export class NftDetectionController extends BaseController<
-          } = nft.token;
-
-          // Use a fallback if metadata is null
-          const { imageOriginal: imageOriginalUrl } = metadata || {};
-
-          let ignored;
-          /* istanbul ignore else */
-          const { ignoredNfts } = this.#getNftState();
-          if (ignoredNfts.length) {
-            ignored = ignoredNfts.find((c) => {
-              /* istanbul ignore next */
-              return (
-                c.address === toChecksumHexAddress(contract) &&
-                c.tokenId === tokenId
-              );
-            });
-          }
+      // Add NFTs to state
+      await this.#addNftsToState(apiNfts, userAddress);
 
-          /* istanbul ignore else */
-          if (!ignored) {
-            /* istanbul ignore next */
-            const nftMetadata: NftMetadata = Object.assign(
-              {},
-              { name },
-              description && { description },
-              imageUrl && { image: imageUrl },
-              imageThumbnailUrl && { imageThumbnail: imageThumbnailUrl },
-              imageOriginalUrl && { imageOriginal: imageOriginalUrl },
-              kind && { standard: kind.toUpperCase() },
-              lastSale && { lastSale },
-              attributes && { attributes },
-              topBid && { topBid },
-              rarityRank && { rarityRank },
-              rarityScore && { rarityScore },
-              collection && { collection },
-              chainId && { chainId },
-            );
-            await this.#addNft(contract, tokenId, {
-              nftMetadata,
-              userAddress,
-              source: Source.Detected,
-              chainId: toHex(chainId),
-            });
-          }
-        });
-        await Promise.all(addNftPromises);
-      } while ((next = resultNftApi.continuation));
       updateSucceeded();
+      console.log(
+        '[NftDetectionController] NFT detection completed successfully',
+      );
+
+      return {
+        nfts: apiNfts,
+        continuationToken: resultNftApi.continuation,
+        hasMore: Boolean(resultNftApi.continuation),
+        totalProcessed: apiNfts.length,
+      };
     } catch (error) {
+      console.error(
+        '[NftDetectionController] Error during NFT detection:',
+        error,
+      );
       updateFailed(error);
       throw error;
     } finally {
       delete this.#inProcessNftFetchingUpdates[updateKey];
     }
   }
+
+  /**
+   * Loads more NFTs using a continuation token from a previous detection.
+   * This is used for pagination/infinite scrolling scenarios.
+   *
+   * @param chainIds - The chain IDs to detect NFTs on.
+   * @param continuationToken - The token from a previous detection response.
+   * @param options - Options bag.
+   * @param options.userAddress - The address to detect NFTs for.
+   * @returns Promise resolving to an object containing the NFTs found, whether there are more to load,
+   * and a continuation token for the next request.
+   */
+  async detectMoreNfts(
```

Comment:
Looks good! perhaps if we go eventually with this approach, we wont need to have two public functions detectMoreNfts and detectNfts;
I imagined we could just update the detectNfts fct to not have a loop to go through all cursors, instead only fetch a page at a time when client requests it